packages/suite/src/views/wallet/trading/common/TradingForm/TradingFormInput/TradingFormInputCryptoSelect/TradingFormInputCryptoSelect.tsx

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useState } from 'react';
+import { Controller } from 'react-hook-form';
+
+import {
+    TradingBuyFormProps,
+    TradingExchangeFormProps,
+    selectTradingLoadingAndTimestamp,
+} from '@suite-common/trading';
+import { Select } from '@trezor/components';
+
+import { Translation } from 'src/components/suite/Translation';
+import { useSelector, useTranslation } from 'src/hooks/suite';
+import { TradingFormInputCryptoSelectProps } from 'src/types/trading/tradingForm';
+
+import { TradingSelectAssetModal } from './TradingSelectAssetModal/TradingSelectAssetModal';
+
+export const TradingFormInputCryptoSelect = <
+    TFieldValues extends TradingBuyFormProps | TradingExchangeFormProps,
+>({
+    label,
+    cryptoSelectName,
+    supportedCryptoCurrencies,
+    methods: { control },
+    isDisabled,
+    placeholder,
+    'data-testid': dataTestId,
+    sortTokensByFiatBalanceInDesc,
+}: TradingFormInputCryptoSelectProps<TFieldValues>) => {
+    const [isModalActive, setIsModalActive] = useState(false);
+    const closeModal = useCallback(() => setIsModalActive(false), []);
+
+    const { translationString } = useTranslation();
+    const { isLoading } = useSelector(selectTradingLoadingAndTimestamp);
+
+    return (
+        <>
+            {isModalActive && (
+                <TradingSelectAssetModal
+                    supportedCryptoCurrencies={supportedCryptoCurrencies}
+                    onModalClose={closeModal}
+                    dataTestId={dataTestId}
+                    sortTokensByFiatBalanceInDesc={sortTokensByFiatBalanceInDesc}
+                />
+            )}
+
+            {/* TODO: Select not needed  */}
+            <Controller
+                name={cryptoSelectName}
+                control={control}
+                render={({ field: { value } }) => (
+                    <Select
+                        placeholder={placeholder && !isLoading && translationString(placeholder)}
+                        value={value}
+                        // The select isn't used. The modal above opens on click/focus event.
+                        options={[]}
+                        labelLeft={label && <Translation id={label} />}
+                        onMenuOpen={() => setIsModalActive(true)}
+                        data-testid={dataTestId ?? '@trading/form/select-crypto'}
+                        isClearable={false}
+                        isMenuOpen={false}
+                        isDisabled={isDisabled || isLoading}
+                        isLoading={isLoading}
+                    />
+                )}
+            />
+        </>
```

Comment:
Remove some redundant props. As the TODO said (added by Tomas Klima sometime ago), the Select input isn't need. However I didn't have time to replace it at the moment.